en/xml/structure/topic.preserve-entities.xml

```diff
@@ -15,32 +15,212 @@
+      </step>
+      <step>
+        <para>Download the Python script <filename
+          xlink:href="#li.structure.preserve-entities.ents2text"
+          >ents2text.py</filename><!--
+          from <link
+            xlink:href="https://github.com/tomschr/dbcookbook/raw/develop/en/xml/structure/entities/ents2text.py"/>
+          -->.
+        </para>
+      </step>
+      <step>
+        <para>Save the script where it can be found by the system. Linux users
+          can store the script in either <filename class="directory"
+            >~/bin</filename> or <filename class="directory"
+          >/usr/bin</filename>. In this procedure, we use <filename
+            class="directory">~/bin</filename>. Make the script executable
+          with:</para>
+        <screen><prompt>$ </prompt><command>chmod</command> +x ~/bin/ents2text.py</screen>
+      </step>
+      <step>
+        <para>Create a link:</para>
+        <screen><prompt>$ </prompt><command>ln</command> -rs ~/bin/ents2text.py ~/bin/text2ents.p</screen>
+        <para>This is necessary to use the conversion from entities to text and
+          vice versa. The script decides the mode by its script name.</para>
+      </step>
+      <step>
+        <para>Convert all entities to text with the following command:</para>
+        <screen><prompt>$ </prompt><command>ents2text.py</command> XMLFILE1 XMLFILE2...</screen>
+        <para>The script converts all entities to their protected notation in
+          place and creates backup files with the extension <filename
+            class="extension">.bak</filename>.</para>
+      </step>
+      <step>
+        <para>Process your XML file with your XML tools.</para>
+      </step>
+      <step>
+        <para>Revert all the protected notation to their original entity
+          notation:</para>
+        <screen><prompt>$ </prompt><command>text2ents.py</command> XMLFILE1 XMLFILE2...</screen>
+      </step>
+    </procedure>
   </section>
   <section role="discussion">
     <title>Discussion</title>
-    <para>Some discussion about problem and solution.</para>
+    <para><emphasis>Not</emphasis> expanding entities looks like
+      an abnormal use case as XML parsers are supposed to expand entities by
+      default.
+      However, expanding entities has one disadvantage: after the XML parser
+      has resolved entities, the content is indistinguishable from other
+      content. In other words, you cannot distinguish content that comes
+      from an entity definition from existing content. You cannot
+      <quote>go back</quote> and revert this process once all entities are
+      expanded.
+    </para>
+    <para>
+      For this reason, preserving entities can be useful when you want to
+      process your XML, but keep your entities untouched. <xref
+        linkend="pro.structure.preserve-entities.workflow"
+        xrefstyle="select:label"/> showed one solution. However, you should
```

Comment:
~~~you should~~~ be aware of certain issues that might cause problems with XML files